### PR TITLE
fix: keyword classifier overrides

### PR DESCRIPTION
**Pull request by wochinge**
**Proposed changes**:
- fix https://github.com/RasaHQ/rasa/issues/11540

**Status (please check what you already did)**:
- [x] added some tests for the functionality
- [ ] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: changelog/11540.bugfix.md [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Fix bug where `KeywordIntentClassifier` overrides `DIETClassifier` predictions
```
Comment (twerkmeister) [inline changelog/11540.bugfix.md:1]:
```suggestion
Fix bug where `KeywordIntentClassifier` overrides preceding intent classifiers' predictions
```

**Review by twerkmeister — APPROVED** [review]
Looking very good! It's a pleasure to see a PR from a familiar face 😎

**Comment by wochinge** [comment]
A bit of nostalgia @twerkmeister 💔 You'll have to press merge for me these days 😁

**Comment by wochinge** [comment]
Haha, let me know if you have some graph stuff 😁

**Pull request merged**